### PR TITLE
[Movies2Watch] Add new extractor

### PR DESCRIPTION
**Pull request by GNtrazios**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [x] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [x] Covered the code with tests (note that PRs without tests will be REJECTED)
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

The purpose of this pull request is to add a new extractor to the list. This PR contains the extractor of Movies2Watch website.

### CODE REVIEW

File: youtube_dl/extractor/movies2watch.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    def _real_extract(self, url):
+
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        title = self._html_search_regex(r'<h1 itemprop="name" class="title">(.+?)</h1>', webpage, 'title')
+        description = self._html_search_regex(r'<div itemprop="description" class="desc shorting" data-type="text">(.+?)</div>', webpage, 'description')
+
+        return {
+            'url': url,
```
Comment (dirkf) [inline youtube_dl/extractor/movies2watch.py:39]: The same URL that was passed in!

You actually have to do some extraction to get the media URL. Did you run this code?

Also relax the regexes, but improving metadata extraction is not the first priority.

**Comment by MrRawes** [comment]
https://github.com/ytdl-org/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free

**Comment by GNtrazios** [comment]
At first thank you for your time.
It's my first trial and I wanna get some time to improve my code in order to have the result that I want

**Comment by CHJ85** [comment]
I'd like for the tv shows to work as well, so I fixed the URL structure and added a test example with not much luck. Although at least movies that end with 1-1 work now:
I'll post my work in progress here in case somebody wants to give it a world.

```
# coding: utf-8
from __future__ import unicode_literals

from .common import InfoExtractor


class Movies2WatchIE(InfoExtractor):
    _VALID_URL = r'https?://movies2watch\.ru/(?:movie|tv)/(?P<id>[^/?#&]+)/(?:\d+-\d+|1-full)$'
    _TESTS = [
        {
            'url': 'https://movies2watch.ru/movie/double-threat-wqyq6/1-full',
            'md5': 'c4ce357bf745d4d27ef7f3b94c9a5dc9',
            'info_dict': {
                'id': 'double-threat-wqyq6',
                'ext': 'mp4',
                'title': 'Double Threat',
                'description': 'After skimming money from the mob, a beautiful young woman finds herself on the run with a kind stranger on a pilgrimage across the country to scatter his brother\'s ashes. In the heat of the moment, we quickly learn that her split personality comes in handy as the ruthless, dynamic side of her is unstoppable.',
            },
        },
        {
            'url': 'https://movies2watch.ru/tv/hey-arnold-n3rxl/1-1',
            'info_dict': {
                'id': 'hey-arnold-n3rxl',
                'ext': 'mp4',
                'title': 'Hey Arnold',
                'description': 'The everyday life of Arnold, a fourth-grader in a nameless city that resembles Brooklyn, New York, who lives in a multi-racial boarding house with his grandparents and a motley assortment of friends and neighbors.',
            },
        },
        {
            'url': 'https://movies2watch.ru/movie/the-batman-j2lx4/1-full',
            'md5': 'a6824ac8f96cdbf839a258493384ea5e',
            'info_dict': {
                'id': 'the-batman-j2lx4',
                'ext': 'mp4',
                'title': 'The Batman',
                'description': 'Two years of nights have turned Bruce Wayne into a nocturnal animal. But as he continues to find his way as Gotham\'s dark knight Bruce is forced into a game of cat and mouse with his biggest threat so far, a manic killer known as "The Riddler" who is filled with rage and determined to expose the corrupt system whilst picking off all of Gotham\'s key political figures. Working with both established and new allies, Bruce must track down the killer and see him brought to justice, while investigating his father\'s true legacy and questioning the affect that he has had on Gotham so far as "The Batman."',
            },
        },
    ]

    def _real_extract(self, url):
        video_id = self._match_id(url)
        webpage = self._download_webpage(url, video_id)

        title = self._html_search_regex(r'<h1 itemprop="name" class="title">(.+?)</h1>', webpage, 'title')
        description = self._html_search_regex(r'<div itemprop="description" class="desc shorting" data-type="text">(.+?)</div>', webpage, 'description')

        return {
            'url': url,
            'id': video_id,
            'title': title,
            'description': description,
            'ext': 'mp4',
        }

```

**Comment by dirkf** [comment]
Code like this can't work unless the URL passed to the extractor is already a video link, in which case it won't have a `title` or `description` inside HTML. The `url` or a list of `formats` has to be extracted in some way as well.

All of this is documented in the manual but passed over the head of the PR author -- which is why the PR is closed.

**Pull request closed without merging**